File: w3/w3java/Paper.java
public class Paper {
    private StringBuffer content;
    private int maxSymbols;
    private int symbols;
    
    public Paper() {
        this.maxSymbols = 4000;
        this.symbols = 0;
        content = new StringBuffer();
    }
    
    public Paper(int maxSymbols) {
        this.maxSymbols = maxSymbols;
        this.symbols = 0;
        content = new StringBuffer();
    }
    
    public int getMaxSymbols() {
        return maxSymbols;
    }
    
    public int getSymbols() {
        return symbols;
    }
    
    public void addContent(String message) throws OutOfSpaceException {
        int total = content.length() + message.length();
        
        if ( content.length() == maxSymbols ) {
            throw new OutOfSpaceException();
        }
        if ( total > maxSymbols ) {
            content.append(message.substring(0, maxSymbols-content.length()));
        }
        content.append(message);
    }
    
    public void show() {
        System.out.println(content);
    }
}

File: w4/PHP/Classes/Paper.php
<?php
    class Paper {
        private $maxSymbols;
        private $symbols;
        private $content;

        private function validate($value) {
            if ( is_numeric($value) ) {
                return $value;
            }
            throw new Exception("Invalid param");
        }

        public function __construct($maxSymbols=64) {
            $this->maxSymbols = $this->validate($maxSymbols);
            $this->symbols = 0;
            $this->content = "";
        }

        public function __get($name) {
            if ( property_exists($this, $name) ) {
                return $this->$name;
            }
            throw new Exception("Attribute error: attribute $name not found");
        }

        public function addContent(String $message) {
            if ( strlen($this->content) == $this->maxSymbols ) {
                throw new Exception("Out of space");
            }

            $this->content = str_pad($this->content, strlen($this->content) + strlen($message), $message);
            $this->symbols = strlen($this->content);
        }

        public function show() {
            echo $this->content . PHP_EOL;
        }
    }
?>